flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -304,6 +305,18 @@ private State<T> createMiddleStates(final State<T> sinkState) {
                 if (currentPattern.getQuantifier().getConsumingStrategy()
                         == Quantifier.ConsumingStrategy.NOT_FOLLOW) {
                     // skip notFollow patterns, they are converted into edge conditions
+                    if (currentPattern.getWindowTime() != null
```

Comment:
should use `windowTime` instead of `currentPattern.getWindowTime()`. This also means that we need to calculate `windowTime` in advance.